src/modules/uac_redirect/rd_funcs.c

```diff
@@ -294,8 +295,16 @@ static int shmcontact2dset(struct sip_msg *req, struct sip_msg *sh_rpl,
 		LM_DBG("adding contact <%.*s>\n", scontacts[i]->uri.len,
 				scontacts[i]->uri.s);
 		if(sruid_next(&_redirect_sruid)==0) {
+			if (!scontacts[i]->flags || str2int(&(scontacts[i]->flags->body), &flags) < 0) {
+				if (flags_hdr_mode && scontacts[i]->flags && str2int(&(scontacts[i]->flags->body), &flags) == 0) {
+					if (flags_hdr_mode == 2)
+						flags |= bflags;
+				} else {
+					flags = bflags;
+				}
+			}
 			if(append_branch( 0, &scontacts[i]->uri, 0, 0, sqvalues[i],
-						bflags, 0, &_redirect_sruid.uid, 0,
+						flags, 0, &_redirect_sruid.uid, 0,
```

Comment:
As I got it, if flags parameter in contact is set and converted successfully to integer, even `flags_hdr_mode == 2`, the bflags is not taken in consideration.

Also I don't see how `flags_hdr_mode == 0` makes use of `bflags` like it was before for backward compatibility when `flags` param in contact is set.

Reply:
1. If conversion is successful, flags are OR'ed with bflags when mode == 2.

There is something wonky here, looks like an incorrectly resolved merge/rebase. I'll fix this up.

2. Good catch, i'll fix this.

Reply:
Both issues were caused by the same broken merge/rebase. Solved in my new patchset.